### PR TITLE
Grt python

### PR DESCRIPTION
**Pull request by macd**
Not all the tcl tests have a Python version yet, but there is a good variety of Python regressions.

### CODE REVIEW

**Comment by macd** [comment]
The failure in format code looks to be a failure running on a .i file. That is probably not what we want to do.

**Comment by maliberty** [comment]
@vvbandeira @habibayassin .i files should be excluded from clang-fomat as they are not c++

**Comment by maliberty** [comment]
I've excluded .i in https://github.com/The-OpenROAD-Project/OpenROAD/pull/2403 and you'll need to merge master after that PR is merged.

**Comment by macd** [comment]
Sounds good.

File: src/grt/README.md
Comment (maliberty) [inline src/grt/README.md:260]: It would be better if these took a dbLayer* rather than an integer.

File: src/grt/include/grt/GlobalRouter.h
Comment (maliberty) [inline src/grt/include/grt/GlobalRouter.h:156]: Just remove them - same below

File: src/grt/test/grt_aux.py
Comment (maliberty) [inline src/grt/test/grt_aux.py:62]: should be checked in c++

File: src/grt/test/grt_aux.py
Comment (maliberty) [inline src/grt/test/grt_aux.py:118]: check in c++

**Review by maliberty — CHANGES_REQUESTED** [review]

**Comment by maliberty** [comment]
In general it feels like there is too much support code in the unit tests that should be either in c++ or a better api.

File: src/grt/README.md
Comment (macd) [reply src/grt/README.md:260]: So these are defined by the current C++ api (in GlobalRouter.h) that is then used by GlobalRouter.i and then further wrapped by GlobalRouter.tcl. So if we change this in C++ we will have to change the tcl interface too (or at least the tcl code to maintain a similar interface), or we can have a parallel interface (not my preference).

File: src/grt/include/grt/GlobalRouter.h
Comment (macd) [reply src/grt/include/grt/GlobalRouter.h:156]: OK

File: src/grt/test/grt_aux.py
Comment (macd) [reply src/grt/test/grt_aux.py:62]: So effectively, that means moving much of grt_aux.py into the C++ api. You comment below that "In general it feels like there is too much support code in the unit tests..." I can understand that but the "support code" for the tcl interface lies in both the GlobalRouter.i and the GlobalRouter.tcl files and totals about 800 lines (let's de-rate to 700 for boilerplate). The GlobalRouter-py.i file and grt_aux.py total about 200 loc, so that is still quite a step down. It will be better in C++ for sure, but it is a redesign of the api.

**Comment by maliberty** [comment]
Are you wanting to tackle the API redesign or do you want to just merge this and leave it to me to handle later?

**Comment by macd** [comment]
Good question Matt. I think I'd like to merge this and have you do the api redesign later. I don't think I know enough about the whole flow to be able to put down a consistent api at the moment. I will do a few of the simple changes first. Should I leave grt_aux.py in grt/test or should we move it to grt/src/GlobalRouter.py to mirror grt/src/GlobalRouter.tcl? I chose the former to signal "this api might not be stable".

**Comment by maliberty** [comment]
Leave grt_aux.py in test

**Review by maliberty — APPROVED** [review]

**Pull request merged**
